app/models/sequencescape_submission.rb

```diff
@@ -104,28 +104,34 @@ def extra_assets
   #   uuids, indicating an asset group. Keys are ignored.
   #
   def asset_groups=(asset_groups)
-    @asset_groups = asset_groups.respond_to?(:values) ? asset_groups.values : asset_groups
+    groups = asset_groups.respond_to?(:values) ? asset_groups.values : asset_groups
+    @asset_groups =
+      groups.map do |group|
+        group[:asset_uuids] = group[:assets] if group[:assets]
+        group.except(:assets)
+      end
   end
 
   def asset_groups_for_orders_creation
     return asset_groups unless (asset_groups.length == 1) && extra_barcodes
 
-    [{ assets: [assets, extra_assets].flatten.compact, autodetect_studies: true, autodetect_projects: true }]
+    [{ asset_uuids: [assets, extra_assets].flatten.compact, autodetect_studies: true, autodetect_projects: true }]
   end
 
   private
 
   def generate_orders
     asset_groups_for_orders_creation.map do |asset_group|
-      order_parameters = { request_options:, user: }.merge(asset_group)
-      submission_template.orders.create!(order_parameters)
+      Sequencescape::Api::V2::Order.create!(
+        submission_template_attributes: { request_options: request_options, user_uuid: user }.merge(asset_group),
+        submission_template_uuid: template_uuid
+      )
     end
   end
 
   # rubocop:disable Metrics/AbcSize
   def generate_submissions
-    orders = generate_orders
-    submission = api.submission.create!(orders: orders.map(&:uuid), user: user)
+    submission = api.submission.create!(orders: generate_orders.map(&:uuid), user: user)
```

Comment:
Is this still using V1 API ?

Reply:
Yes, but we have Submissions in a different pull request that's already approved: https://github.com/sanger/limber/pull/2124/files#diff-5aa7fa045ceaed31dd518b4b69670d92f5b02e363c46968e71ff7ca1c1194cb3